lms/djangoapps/mobile_api/utils.py

```diff
@@ -37,18 +63,22 @@ def _wrapper(self, request, *args, **kwargs):
     return _decorator
 
 
-def mobile_access_when_enrolled(course, user):
+def mobile_course_listing_access(course, user):
     """
-    Determines whether a user has access to a course in a mobile context.
-    Checks the mobile_available flag and the start_date.
-    Note: Does not check if the user is actually enrolled in the course.
+    Determines whether a user has access to a course' listing in a mobile context.
+        Checks the mobile_available flag.
+        Checks roles including Beta Tester and staff roles.
+    Note:
+        Does not check if the user is actually enrolled in the course.
+        Does not check the start_date.
     """
     # The course doesn't always really exist -- we can have bad data in the enrollments
     # pointing to non-existent (or removed) courses, in which case `course` is None.
     if not course:
         return False
     try:
-        return get_course_with_access(user, 'load_mobile_no_enrollment_check', course.id) is not None
+        with dict_value(settings.FEATURES, 'DISABLE_START_DATES', True):
```

Comment:
I'm not sure what a better way to do this is (plumbing the flag through I guess), but I'm pretty sketched out by overriding a config setting.

If we're stuck with global variables because passing the value through is a mess, maybe we can at least replace this with a global function that indirects access to ``settings.FEATURES``, but using a context variable if its set? I can imagine someone trying to use DISABLE_START_DATES for some reason and being very surprised when its value didn't match the config